### PR TITLE
complete-guide-on-using-sequelize-basic-and-advanced-features

### LINKED ISSUE #2949: Complete Guide on using Sequelize basic and advanced features.
### Proposed title of article
Complete Guide on using Sequelize basic and advanced features.

### Introduction paragraph (2-3 paragraphs):
Sequelize is a JavaScript library that manages database. It supports all the popular databases such as PostgreSQL, MySQL, SQLite and MSSQL. This kind of package is known as an Object Relation Mapper, ORM’s are used to convert object relational syntax to database schema. Sequelize is used with NodeJS.
Sequelize only support relational databases, database like mongo DB or other NoSQL databases are not supported.  Sequelize supports solid transaction, eager and lazy loading, relations, read replication and other cool features.

### Key takeaways: 

- How to install sequelize
- How to install drivers associated with the different database
- Connecting to the database and testing the connection
- How to use Sequelize CLI
- Model Querying (Inserting, Updating, Deleting)
- Associations (One to One, One to Many, Many to Many).

### References: 
[Manual | Sequelize](https://sequelize.org/master/index.html)

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Josh4324**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #2949

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/complete-guide-on-using-sequelize-basic-and-advanced-features/index.md [hunk 1]
```diff
@@ -0,0 +1,1102 @@
...
+- A basic kowledge of Node Js and Express Js.
+- Familarization with any of these databases i.e MYSQL, POSTGRESQL, MSSQL, SQLITE
+- A basic knowledge of postman, which we will use to test our endpoints.
+- Text Editor - i will be using VS Code.
+
+In this article, I will show you most features you will need when using sequelize, and i will be using a project based approach.
+it is important you follow along and practise the concepts.
+
+Let's get started!
+
```
[the remark this answers is not in the document]
Comment (Josh4324) [reply content/articles/complete-guide-on-using-sequelize-basic-and-advanced-features/index.md:29]: @srishilesh so i covered both the ORM and sequelize in the definition, the point of this article is to focus more on the practical side, there are already too many articles that explains what an ORM or sequelize does in details.

**Comment by Josh4324** [comment]
@srishilesh Do i need to create a duplicate PR since this is closed

**Comment by srishilesh** [comment]
> @srishilesh Do i need to create a duplicate PR since this is closed

@Josh4324 

Are the changes done and pushed to your forked branch? If so, can you share the link?

I'll verify and reopen this PR. Once re-opened, you can push the latest changes here.

Thanks

**Comment by Josh4324** [comment]
@srishilesh https://github.com/Josh4324/engineering-education/blob/josh-article/content/articles/complete-guide-on-using-sequelize-basic-and-advanced-features/index.md

**Comment by srishilesh** [comment]
@Josh4324 

You can create a new PR and tag me there.

**Pull request closed without merging**